### PR TITLE
DocumentCollection PublishingApi migration

### PR DESCRIPTION
**Pull request by mgrassotti**
Trello: https://trello.com/c/DoIbAkZB/428-11-document-collection-migration-implement-publishing-of-format-to-publishing-api
- Remove the Placeholder Presenter
- Add a Presenter for the PublishingApi
- Add Sync Checks

mobbed by @andrewgarner, @bevanloon, @binaryberry, @gpeng, @MatMoore, @mgrassotti, @tvararu

### CODE REVIEW

File: app/presenters/publishing_api/document_collection_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        body: govspeak_renderer.govspeak_edition_to_html(item),
+        emphasised_organisations: item.lead_organisations.map(&:content_id),
+      }.tap do |details_hash|
+        details_hash.merge!(PayloadBuilder::PoliticalDetails.for(item))
+        details_hash.merge!(PayloadBuilder::AccessLimitation.for(item))
+      end
+    end
+
+    def first_public_at
+      item.first_published_at || item.document.created_at
```
Comment (MatMoore) [inline app/presenters/publishing_api/document_collection_presenter.rb]: Did we decide whether to move this method to `Edition`? I think it's worth extracting somewhere as we're duplicating the logic in all the presenters at the moment.

Comment (bevanloon) [inline app/presenters/publishing_api/document_collection_presenter.rb]: We've decided to leave this for now as `first_public_at` is already exposed on Edition with different functionality and is used elsewhere. There are cards in Trello https://trello.com/c/j7B7RSMa/729-migrated-formats-no-longer-need-to-send-first-public-at and https://trello.com/c/YhI80ytY/727-ensure-migrated-formats-are-rendered-using-first-published-at to address this.

File: lib/sync_checker/formats/document_collection_check.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      def rendering_app
+        Whitehall::RenderingApp::WHITEHALL_FRONTEND
+      end
+
+      def checks_for_live(locale)
+        super + [
+          Checks::LinksCheck.new(
+            "topical_events",
+            TopicalEvent
+            .for_edition(edition_expected_in_live.id)
```
Comment (boffbowsh) [inline lib/sync_checker/formats/document_collection_check.rb]: These two should be indented an extra level

Comment (gpeng) [inline lib/sync_checker/formats/document_collection_check.rb]: Fixed in 77e46561

**Pull request merged**